clients/ts/signalr/src/HubConnection.ts

```diff
@@ -316,11 +316,16 @@ export class HubConnection {
             this.methods[methodName] = [];
         }
 
-        this.methods[methodName].push(method);
+        // Preventing adding the same handler multiple times.
+        for (const method of this.methods[methodName]) {
```

Comment:
You can use `indexOf`, it'll return `-1` if the item isn't in the list. So `if(this.methods.indexOf(newMethod) < 0) { return; }`

Reply:
I was unsure if indexOf used `===` but looks like it does 

>indexOf() compares searchElement to elements of the Array using strict equality (the same method used by the === or triple-equals operator).

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf